middlewares/retry.go

```diff
@@ -32,12 +33,19 @@ func NewRetry(attempts int, next http.Handler) *Retry {
 }
 
 func (retry *Retry) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	// if we might make multiple attempts, swap the body for an ioutil.NopCloser
+	// cf https://github.com/containous/traefik/issues/1008
+	if retry.attempts > 1 {
+		body := r.Body
+		defer body.Close()
+		r.Body = ioutil.NopCloser(body)
+	}
 	attempts := 1
 	for {
 		recorder := NewRecorder()
 		recorder.responseWriter = rw
 		retry.next.ServeHTTP(recorder, r)
-		if !isNetworkError(recorder.Code) || attempts >= retry.attempts {
+		if recorder.Error == nil || attempts >= retry.attempts {
```

Comment:
>If this field is not nil then it comes from the transport RoundTrip call, meaning it is a network error.

I'm really not sure about that. In oxy, `err` is not null at multiple places:
- https://github.com/containous/oxy/blob/master/forward/fwd.go#L164
- https://github.com/containous/oxy/blob/master/forward/fwd.go#L198
- https://github.com/containous/oxy/blob/master/forward/fwd.go#L265
- https://github.com/containous/oxy/blob/master/forward/fwd.go#L278
- https://github.com/containous/oxy/blob/master/forward/fwd.go#L288